src/commands/treeCommand.ts

```diff
@@ -0,0 +1,72 @@
+import * as vscode from "vscode";
+import { Logs } from "../models/logs";
+import {
+  commands
+} from "../utils/common/commands";
+import { constants } from "../utils/common/constants";
+import { AstResultsProvider } from "../views/resultsView/astResultsProvider";
+import { SCAResultsProvider } from "../views/scaView/scaResultsProvider";
+
+export class TreeCommand {
+  context: vscode.ExtensionContext;
+  astResultsProvider: AstResultsProvider;
+  scaResultsProvider: SCAResultsProvider;
+  logs: Logs;
+  constructor(
+    context: vscode.ExtensionContext,
+    astResultsProvider: AstResultsProvider,
+    scaResultsProvider: SCAResultsProvider,
+    logs: Logs
+  ) {
+    this.context = context;
+    this.astResultsProvider = astResultsProvider;
+    this.scaResultsProvider = scaResultsProvider;
+    this.logs = logs;
+  }
+
+  public registerRefreshCommands() {
+    this.createRefreshTreeCommand();
+    this.createRefreshScaTreeCommand();
+  }
+
+  public registeClearCommands() {
+    this.clearCommand();
+    this.clearScaCommand();
+  }
+
+  private createRefreshTreeCommand() {
+    this.context.subscriptions.push(
+      vscode.commands.registerCommand(
+        commands.refreshTree,
+        async () => await this.astResultsProvider.refreshData()
```

Comment:
@tiagobcx  doesn't seem that this class holds the logic, most of the things are still done inside the providers. So maybe put it together.

Reply:
This commands classes were supposed to have the register commands or some specific actions like for groups or filters. Only way to join this would be to move all this register commands inside the results and sca providers. Is that what you meant?